src/python/gem5/components/processors/spatter_gen/spatter_kernel.py

```diff
@@ -0,0 +1,181 @@
+# Copyright (c) 2024 The Regents of the University of California
+# All rights reserved.
+#
+# Redistribution and use in source and binary forms, with or without
+# modification, are permitted provided that the following conditions are
+# met: redistributions of source code must retain the above copyright
+# notice, this list of conditions and the following disclaimer;
+# redistributions in binary form must reproduce the above copyright
+# notice, this list of conditions and the following disclaimer in the
+# documentation and/or other materials provided with the distribution;
+# neither the name of the copyright holders nor the names of its
+# contributors may be used to endorse or promote products derived from
+# this software without specific prior written permission.
+#
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+# "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+# LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+# A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+# OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+# SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+# LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+# DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+# THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+# (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+# OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+from math import ceil
+from typing import (
+    List,
+    Tuple,
+)
+
+from m5.objects import SpatterKernelType
+from m5.params import Addr
+
+
+def parse_kernel(kernel: dict) -> Tuple[int, int, str, List]:
+    delta = kernel.get("delta", 0)
+    count = kernel.get("count", 1)
+    type = kernel.get("kernel", None)
+    if type is None:
+        raise ValueError(f"Keyword 'kernel' not found.")
+    type = SpatterKernelType(type.lower())
+    trace = kernel.get("pattern", [])
+    if len(trace) == 0:
+        raise ValueError(f"Empty 'pattern' found.")
+    return (delta, count, type, trace)
+
+
+def partition_trace(original_trace, num_partitions, interleave_size):
+    partitions = [[] for _ in range(num_partitions)]
+    num_leaves = ceil(len(original_trace) / interleave_size)
+    for i in range(num_leaves):
+        lower_bound = i * interleave_size
+        upper_bound = min(lower_bound + interleave_size, len(original_trace))
+        partitions[i % num_partitions] += original_trace[
+            lower_bound:upper_bound
+        ]
+    return partitions
+
+
+class SpatterKernel:
+    """This class encapsulates one kernel in a spatter trace.
+    A spatter trace is represented with a json file.
+    An example of a spatter trace can be found here:
+        bit.ly/3X2IijM
```

Comment:
Can you point to the repo instead? I don't trust that bitly will be around in 20 years.